src/decoder/symphonia.rs

```diff
@@ -197,7 +202,7 @@ impl Source for SymphoniaDecoder {
         let seek_res = self
             .format
             .seek(
-                SeekMode::Accurate,
```

Comment:
GitHub wont let me leave a comment where I want to.. So this is about `skip_back_a_tiny_bit`.

Like you said in #687 we should make it so the source returns None next if the seek is beyond the end. That might still fit in this PR.

Reply:
Removed that `skip_back_a_tiny_bit` and threw in another small optimization with getting `total_duration`.

I was thinking of doing quite some more refactoring of the Symphonia decoder, but now think it's better to get this "builder" PR settled. To do meaningful refactoring there, I need to get this span length thing down first anyway.

Reply:
agreed, lets do that later :+1: